oak_runtime/src/lib.rs

```diff
@@ -575,9 +588,15 @@ impl Runtime {
     fn get_node_downgraded_label(&self, node_id: NodeId, initial_label: &Label) -> Label {
         // Retrieve the set of tags that the node may downgrade.
         let node_privilege = self.get_node_privilege(node_id);
-        Label {
-            // Remove all the confidentiality tags that the Node may declassify.
-            confidentiality_tags: initial_label
+
+        // Remove all the confidentiality tags that the Node may declassify.
+        // TODO(#1631): Perform this check for each sub-lattice.
+        let confidentiality_tags = match node_privilege
+            .can_declassify_confidentiality_tags
+            .contains(&top())
+        {
+            true => vec![],
+            false => initial_label
```

Comment:
I'm not a Rust programmer, so I might just not know Rust conventions. For me this would be slightly more intuitive if it was an if/else expression to check for top and then a match within the else (not top) branch.

The way you have it already is fine, though.

Reply:
Thanks. Changed it to an if-else expression.